ddtrace/internal/datadog/profiling/dd_wrapper/src/interface.cpp

```diff
@@ -139,18 +161,26 @@ ddup_init()
         sigaction(SIGSEGV, &(sigaction_handlers), NULL);
 #endif
 
-        g_profile_real[0] = profile_builder.build_ptr();
-        g_profile_real[1] = profile_builder.build_ptr();
-        g_profile = g_profile_real[g_prof_flag];
-        g_uploader = uploader_builder.build_ptr();
-        is_initialized = true;
+        // install the ddup_fork_handler for pthread_atfork
+        // Right now, only do things in the child _after_ fork
+        pthread_atfork(nullptr, nullptr, ddup_fork_handler);
+
+        // Set the global initialization flag
+        is_ddup_initialized = true;
+        return true;
+    }();
+
+    if (initialized)
+        ++initialized_count;
```

Comment:
does this need to be atomic?

Reply:
It should be, yeah.  The purpose of this value is just to report on the pathological scenario where initialization somehow happens multiple times.  From that perspective, mission accomplished no matter if we shed _extra_ increments every now and again.  However, the precise number could be significant for analysis, in which case we definitely don't want to drop it.

Thank you, I'll fix this.